src/custom/components/SearchModal/CurrencySearch/useTokenSearch.ts

```diff
@@ -0,0 +1,74 @@
+import { Token } from '@uniswap/sdk-core'
+import { atom, useAtom } from 'jotai'
+import { loadable } from 'jotai/utils'
+import { useEffect, useMemo } from 'react'
+
+const BASE_URL = 'https://temp.api.uniswap.org/v1/tokens/search?tokenQuery='
+
+type Address = `0x${string}`
+
+interface FetchTokensResult {
+  chainId: number
+  name: string
+  address: Address
+  decimals: number
+  symbol: string
+  logoURI: string
+  coinGeckoId: string
+  priceUsd: number
+  price24hChange: number
+  volume24h: number
+  marketCap: number
+}
+
+interface FetchTokensApiResult {
+  code: number
+  status: 'success' | 'failed'
+  hasNext: boolean
+  data: FetchTokensResult[]
+}
+
+async function fetchTokens(query: string): Promise<FetchTokensApiResult> {
+  const response = await fetch(`${BASE_URL}${query}`)
+  const data = await response.json()
+
+  return data
```

Comment:
What happens if the request fails (5xx, 4xx) or the response does not contain a json?

Also, no need for the intermediate `data` variable, can return `await response.json()` directly.

Reply:
Loadable essentially wraps this for us, where they return state with `hasError` instead of `hasData`.

Currently I am treating it as if we can't get it, just stay silent (as this is just an external source) - which seems to be the most graceful way of failing there.

However if we'd like to we could rely on the state variable from the loadable atom to craft anything pretty much.

> Also, no need for the intermediate data variable, can return await response.json() directly.
Replacing the data variable, great point.